src/libs/ToastNotifications.tsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import { createRoot } from 'react-dom/client';
+import { AlertColor, Alert, Snackbar, SnackbarOrigin } from '@mui/material';
+
+type ToastPosition = 'topLeft' | 'topRight' | 'bottomLeft' | 'bottomRight';
+
+interface NotificationProps {
+  severity: AlertColor;
+  text: string;
+  timeout: number;
+  layout: ToastPosition | SnackbarOrigin;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  [key: string]: any;
+}
+
+const defaultProps: NotificationProps = {
+  severity: 'info',
+  text: 'default',
+  timeout: 3500,
+  layout: {
+    vertical: 'bottom',
+    horizontal: 'right'
+  }
+};
+
+const convertToSnackbarOrigin = (layout: ToastPosition | SnackbarOrigin): SnackbarOrigin => {
+  if (typeof layout === 'string') {
+    const positions: Record<ToastPosition, SnackbarOrigin> = {
+      topLeft: { vertical: 'top', horizontal: 'left' },
+      topRight: { vertical: 'top', horizontal: 'right' },
+      bottomLeft: { vertical: 'bottom', horizontal: 'left' },
+      bottomRight: { vertical: 'bottom', horizontal: 'right' }
+    };
+    return positions[layout];
+  }
+  return layout;
+};
+
+export const ToastNotifications = {
+  showNotification: ({
+    severity = defaultProps.severity,
+    text = defaultProps.text,
+    timeout = defaultProps.timeout,
+    layout = defaultProps.layout,
+    ...props
+  }: NotificationProps): void => {
+    const snackbarLayout = convertToSnackbarOrigin(layout);
+    const notificationRoot = document.createElement('div');
+    document.body.appendChild(notificationRoot);
+    const root = createRoot(notificationRoot);
```

Comment:
This ensure that the toast pops up in the viewport as opposed to the element and that `react` knows about it.